### PR TITLE
Allow user to input list of directories for `root_dir`

### LINKED ISSUE #15: Tool to combine catalogs into an archive
I think a YAML file like

```
experiment1:
  case1:
    catalog_path: [path to catalog]
    member_id: ###
  case2:
    catalog_path: [path to catalog]
    member_id: ###
experiment2:
  case3:
    catalog_path: [path to catalog]
    member_id: ###
```

would define the ensemble well enough. Each individual catalog would be read in, the experiment and member_id columns would be added, and the `ctrl_case` column would be replaced with `ctrl_experiment` and `ctrl_member_id` (assuming `ctrl_case` is also a member of the ensemble). Then all the individual catalogs would be concatenated into one giant file (we'd need to replace relative path names with absolute ones at this stage).

### LINKED ISSUE #64: Add an "include_patterns" arguement
When building intake-esm catalogs from multiple directories (different cases), it would be helpful to be able to specify which cases to include... we could do this by including an `include_patterns` argument

### PR DESCRIPTION
**Pull request by mgrover1**
Closes #64, possibly #75

### CODE REVIEW

File: ecgtools/builder.py [hunk 1]
```diff
@@ -101,7 +101,16 @@ def get_directories(self):
         `ecgtools.Builder`
         """
         pattern = '*/' * (self.depth + 1)
-        dirs = [x for x in self.root_path.glob(pattern) if x.is_dir()]
+
+        if type(self.root_path) is pathlib.PosixPath:
```
Comment (andersy005) [inline ecgtools/builder.py:105]:
```suggestion
        if isinstance(self.root_path, pathlib.PosixPath):
```

File: ecgtools/builder.py [hunk 2]
```diff
@@ -101,7 +101,16 @@ def get_directories(self):
...
+            dirs = [x for x in self.root_path.glob(pattern) if x.is_dir()]
+
+        elif type(self.root_path) is list:
```
Comment (andersy005) [inline ecgtools/builder.py:108]:
```suggestion
        elif isinstance(self.root_path, list):
```

File: ecgtools/builder.py [hunk 3]
```diff
@@ -101,7 +101,16 @@ def get_directories(self):
...
+            dirs = [x for path in self.root_path for x in path.glob(pattern) if x.is_dir()]
+
+        else:
+            raise ValueError(f'`root_path` must a list or a string. Got {type(self.root_path)}')
```
Comment (andersy005) [inline ecgtools/builder.py:112]: Pydantic's validation should catch during the input validation step... So, I don't think this is needed (I could be wrong though... Double check)

Comment (mgrover1) [reply ecgtools/builder.py:112]: Ahh you are right... it catches it upon read in! Sweet!

**Review by andersy005 — APPROVED** [review]

**Pull request merged**